src/aks-preview/azext_aks_preview/_validators.py

```diff
@@ -213,6 +213,20 @@ def validate_eviction_policy(namespace):
             raise CLIError("--eviction-policy can only be Delete or Deallocate")
 
 
+def validate_spot_max_price(namespace):
+    """Validates the spot node pool max price."""
+    if not isnan(namespace.spot_max_price):
```

Comment:
If spot_max_price is not a number, will it throw an error from model validation in SDK before it reaches server?

Reply:
it will thrown error in CLI.

Reply:
az aks nodepool add: error: argument --spot-max-price: invalid float value: 'fds'